src/libs/actions/IOU.js

```diff
@@ -155,29 +164,73 @@ function requestMoney(report, participants, amount, currency, recipientEmail, de
         debtorEmail,
         preferredLocale,
     );
-    const optimisticData = [
+    let optimisticData = [];
+    let successData = [];
+    let failureData = [];
+
+    // First, let's add the data we need just when we are creating a new chat report
+    if (isNewChat) {
+        const optimisticCreateAction = ReportUtils.buildOptimisticCreatedReportAction(recipientEmail);
+        optimisticData = [
+            {
+                onyxMethod: CONST.ONYX.METHOD.SET,
+                key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`,
+                value: {
+                    ...chatReport,
+                    isLoadingReportActions: true,
+                    lastVisitedTimestamp: Date.now(),
+                    pendingFields: {createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD},
+                    isOptimisticReport: true,
+                },
+            },
+            {
+                onyxMethod: CONST.ONYX.METHOD.SET,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+                value: optimisticCreateAction,
+            },
+        ];
+
+        successData = [
```

Comment:
Yeah I think this might help with the fact the optimistic actions are showing as optimistic if you go offline -> online -> offline again (in which case it should not be optimistic anymore.